tests/test_tracing_tracer.py

```diff
@@ -302,3 +333,50 @@ def test_threaded_memory(tracer, asserts):
     assert tracer.memory.current_span_id is None
 
 
+def test_tracer_extract_google_trace_headers_if_exist_with_span_and_trace_and_options(tracer):
+    google_headers = {
+        "X-Cloud-Trace-Context": "trace-id/span-id;options"
+    }
+    processed_headers = tracer._extract_google_trace_headers_if_exist(google_headers)
+
+    expected_headers = {
+        "X-B3-SpanId": "span-id",
+        "X-B3-TraceId": "trace-id"
+    }
+    assert processed_headers == expected_headers
+
+
+def test_tracer_extract_google_trace_headers_if_exist_with_span_and_trace_and_no_options_and_remaining_semicolon(
+        tracer):
+    google_headers = {
+        "X-Cloud-Trace-Context": "trace-id/span-id;"
+    }
+    processed_headers = tracer._extract_google_trace_headers_if_exist(google_headers)
+
+    expected_headers = {
+        "X-B3-SpanId": "span-id",
+        "X-B3-TraceId": "trace-id"
+    }
+    assert processed_headers == expected_headers
+
+
+def test_tracer_extract_google_trace_headers_if_exist_with_span_and_trace_and_no_options(tracer):
+    google_headers = {
+        "X-Cloud-Trace-Context": "trace-id/span-id"
+    }
+    processed_headers = tracer._extract_google_trace_headers_if_exist(google_headers)
+
+    expected_headers = {
+        "X-B3-SpanId": "span-id",
+        "X-B3-TraceId": "trace-id"
+    }
+    assert processed_headers == expected_headers
```

Comment:
Nice - I like the descriptive test case names!

As all these tests are quite repetitive it might work out better with a parameterised test or by refactoring the expected_headers into a small factory method to make each case smaller.